packages/mui-base/src/Collapsible/Content/useCollapsibleContent.ts

```diff
@@ -0,0 +1,142 @@
+'use client';
+import * as React from 'react';
+import * as ReactDOM from 'react-dom';
+import { mergeReactProps } from '../../utils/mergeReactProps';
+import { ownerWindow } from '../../utils/owner';
+import { useEnhancedEffect } from '../../utils/useEnhancedEffect';
+import { useEventCallback } from '../../utils/useEventCallback';
+import { useAnimationsFinished } from '../../utils/useAnimationsFinished';
+import { useForkRef } from '../../utils/useForkRef';
+import { useId } from '../../utils/useId';
+import {
+  UseCollapsibleContentParameters,
+  UseCollapsibleContentReturnValue,
+} from './CollapsibleContent.types';
+
+function getComputedStyles(element: HTMLElement) {
+  const containerWindow = ownerWindow(element);
+  return containerWindow.getComputedStyle(element);
+}
+
+/**
+ *
+ * Demos:
+ *
+ * - [Collapsible](https://mui.com/base-ui/react-collapsible/#hooks)
+ *
+ * API:
+ *
+ * - [useCollapsibleContent API](https://mui.com/base-ui/react-collapsible/hooks-api/#use-collapsible-content)
+ */
+function useCollapsibleContent(
+  parameters: UseCollapsibleContentParameters,
+): UseCollapsibleContentReturnValue {
+  const {
+    id: idParam,
+    open,
+    mounted: contextMounted,
+    ref,
+    setContentId,
+    setMounted: setContextMounted,
+  } = parameters;
+
+  const id = useId(idParam);
+
+  const contentRef = React.useRef<HTMLElement | null>(null);
+
+  const heightRef = React.useRef(0);
+  const { current: height } = heightRef;
```

Comment:
Reading `ref` in render may be causing stale data issues? This should likely be moved to state